src/brs/BlockchainProcessorImpl.java

```diff
@@ -975,13 +980,32 @@ private void pushBlock(final Block block) throws BlockNotAcceptedException {
 
           calculatedTotalAmount += transaction.getAmountNQT();
           calculatedTotalFee += transaction.getFeeNQT();
+		  
+		  feeArray[slotIdx] = transaction.getFeeNQT();
+		  slotIdx += 1;
+		  
           digest.update(transaction.getBytes());
         }
 
         if (calculatedTotalAmount > block.getTotalAmountNQT()
             || calculatedTotalFee > block.getTotalFeeNQT()) {
           throw new BlockNotAcceptedException("Total amount or fee don't match transaction totals for block " + block.getHeight());
         }
+
+		if (Burst.getFluxCapacitor().isActive(FeatureToggle.SLOT_FEE_ENFORCING_OPTION_2)) {
+			Arrays.sort(feeArray);
+			for (int slotIdx = 0; slotIdx < feeArray.length; slotIdx++) {
```

Comment:
You seem to be overloading `slotIdx` here which makes this code hard to follow.  Maybe rename `slotIdx` on line `985` to something else?